### PR TITLE
HAWQ-154. Update BUILD_INSTRUCTIONS file for dependencies install method

### PR DESCRIPTION
**Pull request by stanlyxiang**
1. update gcc, gcc-c++, make install method.
2. add epel-release

### CODE REVIEW

**Comment by wangzw** [comment]
Good to me. +1

File: BUILD_INSTRUCTIONS.md [hunk 1]
```diff
@@ -40,7 +51,7 @@ Libraries could be installed through yum.
         libyaml-devel       0.1.1
         flex-devel          2.5.35
 ## Build
-After installed the libraries listed above, get code with command:
+After installed the libraries listed above, get source code:
```
Comment (changleicn) [inline BUILD_INSTRUCTIONS.md]: "After installed the libraries listed above, get source code" => "After the libraries list above are installed, get source code"

Comment (stanlyxiang) [inline BUILD_INSTRUCTIONS.md]: I`ll fix it . Thanks.

**Comment by changleicn** [comment]
looks good.

**Pull request closed without merging**